opm/autodiff/FlowMain.hpp

```diff
@@ -391,6 +391,8 @@ namespace Opm
                 deck_ = parser->parseFile(deck_filename, parseContext);
                 checkDeck(deck_, parser);
                 eclipse_state_.reset(new EclipseState(deck_, parseContext));
+                auto ioConfig = eclipse_state_->getIOConfig();
+                ioConfig->setOutputDir(output_dir_);
```

Comment:
I agree this can be fixed later since it is about the EclipseState API and not this particular call site. However I think the solution involving a mutable member and const accessor is bad here. At least in this context we have non-const access to the EclipseState and don't need it to be const. Are there cases where we must modify a const EclipseState? And would not that probably be buggy?